backend/sites/src/app/resolvers/snapshot/snapshot.resolver.ts

```diff
@@ -128,4 +128,50 @@ export class SnapshotsResolver {
       );
     }
   }
+
+  @Roles({ roles: ['site-admin'], mode: RoleMatchingMode.ANY })
+  @Mutation(() => SnapshotResponse, { name: 'createSnapshotForSites' })
+  async createSnapshotForSites(
+    @Args('inputDto', { type: () => [CreateSnapshotDto] }, new ValidationPipe()) inputDto: CreateSnapshotDto[],
+    @AuthenticatedUser() user:any
+  )
+  {
+     try
+     {
+        if(inputDto)
+        {
+          const isSaved =  this.snapshotsService.createSnapshotForSites(inputDto,user);
+          if(isSaved)
+          {
+            return this.genericResponseProvider.createResponse(
+              'Successfully created snapshots.',
+              201,
+              true,
+            );
+          }
+          else
+          {
+            return this.genericResponseProvider.createResponse(
+              `Failed to create snapshots. `,
+              422,
+              false,
+            );
+          }
+        }
+        else
+        {
+          return this.genericResponseProvider.createResponse(
```

Comment:
this else is when the inputDTO is empty right? so have the error message appropriately

Reply:
done.